### PR TITLE
XWIKI-20972: Delete page children tab dropdown toggler has no accessible name

### PR DESCRIPTION
**Pull request by Sereza7**
**Jira:** https://jira.xwiki.org/browse/XWIKI-20972

EDIT: A commit message https://github.com/xwiki/xwiki-platform/pull/2215/commits/bb81586d9c450d8dae3644557c480331c3370fe5 does not follow best practices, please squash this PR when merging.

## PR Changes
* Added labels for the dropdowns, with an escaped translation.

## Notes
No view changes.
This hsould also fix https://jira.xwiki.org/browse/XWIKI-20991. 
The issues are very similar but were reported differently.
## Tests
Didn't trigger the reported WCAG warnings when running:
* `mvn clean install -f xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-docker/ -Pdocker,integration-tests -Dxwiki.test.ui.wcag=true -Dit.test=DeletePageIT#deletePageWithUsedClass`
* `mvn clean install -f xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-docker/ -Pdocker,integration-tests -Dxwiki.test.ui.wcag=true -Dit.test=NavigationIT#navigateWithKeyboardShortcuts`

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/delete.vm [hunk 1]
```diff
@@ -135,7 +135,10 @@
   <div class="panel $!panelClass">
     <div class="panel-heading">
       $heading
-      <div class="pull-right"><a class="panel-collapse-carret" role="button" data-toggle="collapse" href="#$panelId" aria-expanded="false" aria-controls="$panelId"><span class="caret"></span></a></div>
+      <div class="pull-right"><a class="panel-collapse-carret" role="button" data-toggle="collapse" href="#$panelId" aria-expanded="false" aria-controls="$panelId">
+        <span class="caret"></span>
+        <label class='sr-only'>$escapetool.xml($services.localization.render('core.delete.affectChildren.dropDown.label'))</label>
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/delete.vm:140]: The use of the `<label>`-element is invalid here as `<a>`-elements must not have interactive content as children, and a `<label>` is interactive content.

**Review by michitux — CHANGES_REQUESTED** [review]

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm [hunk 2]
```diff
@@ -110,7 +110,10 @@
   <div class="panel $!panelClass">
     <div class="panel-heading">
       $heading
-      <div class="pull-right"><a class="panel-collapse-carret" role="button" data-toggle="collapse" href="#$panelId" aria-expanded="false" aria-controls="$panelId"><span class="caret"></span></a></div>
+      <div class="pull-right"><a class="panel-collapse-carret" role="button" data-toggle="collapse" href="#$panelId" aria-expanded="false" aria-controls="$panelId">
+        <span class="caret"></span>
+        <label class='sr-only'>$escapetool.xml($services.localization.render('core.delete.affectChildren.dropDown.label'))</label>
```
[the remark this answers is not in the document]
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm:115]: Reference: https://html.spec.whatwg.org/#interactive-content

[the remark this answers is not in the document]
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm:115]: For anchors, there's no need to use a label, we can just put a span with the `sr-only` class inside of it. Its accessible name will follow its content.
displayOfAnAnchorWithASpanAsLabel

[the remark this answers is not in the document]
Comment (michitux) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm:115]: Yes, but why is the translation name displayed instead of the translated string in your screenshot? Is this a bug or just due to how you generated the screenshot?

[the remark this answers is not in the document]
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm]: image
Notice the correct name of the element in the Accessibility tree (minus the fact that I didn't bother getting the translations on my distrib)

[the remark this answers is not in the document]
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm]: Fixed in https://github.com/xwiki/xwiki-platform/pull/2215/commits/bb81586d9c450d8dae3644557c480331c3370fe5

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/delete.vm (hunk 1 above)
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/delete.vm]: Fixed in https://github.com/xwiki/xwiki-platform/pull/2215/commits/bb81586d9c450d8dae3644557c480331c3370fe5

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/delete.vm [hunk 3]
```diff
@@ -161,7 +164,7 @@
         #set ($forceAffectChildren = " checked='checked'")
       #end
       #define($heading)
-        <input name="affectChildren" id="affectChildren" type="checkbox" $!{forceAffectChildren}/><label for="affectChildren">$services.localization.render('core.delete.affectChildren')</label> ($childCount)
+        <input name="affectChildren" id="affectChildren" type="checkbox" $!{forceAffectChildren}/><label for="affectChildren">$escapetool.xml($services.localization.render('core.delete.affectChildren')) ($childCount)</label>
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/delete.vm:167]: Why did you move the children count into the label?

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm [hunk 4]
```diff
@@ -126,7 +129,7 @@
   ## Only propose to restore the batch if it also contains other documents than the current one.
   #if ($batchDeletedDocuments.size() > 1)
     #define($heading)
-      <input name="includeBatch" id="includeBatch" type="checkbox" value="true"/><label for="includeBatch">$services.localization.render('core.restore.includeBatch')</label> ($batchDeletedDocuments.size())
+      <input name="includeBatch" id="includeBatch" type="checkbox" value="true"/><label for="includeBatch">$escapetool.xml($services.localization.render('core.restore.includeBatch'))</label> ($batchDeletedDocuments.size())
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm:132]: Wouldn't it make sense to move the batch size into the label here, too, if you do this for the children count during deletion?

**Review by michitux — CHANGES_REQUESTED** [review]

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm]: > Yes, but why is the translation name displayed instead of the translated string in your screenshot? Is this a bug or just due to how you generated the screenshot?

It's just due to how I generated the screenshot, sorry for the imprecision. I did not get the translations on my live test distribution.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm (hunk 4 above)
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/restore.vm]: Fixed in https://github.com/xwiki/xwiki-platform/pull/2215/commits/7bc1714558c04e6e883b60fc0ba973eb96ed9446
